aiida/orm/nodes/data/structure.py

```diff
@@ -806,7 +819,10 @@ def set_ase(self, aseatoms):
         """
         if is_ase_atoms(aseatoms):
             # Read the ase structure
-            self.cell = aseatoms.cell
+            if calc_cell_volume(aseatoms.cell) == 0.0:  # ASE defaults to cell [0.,0.,0.]
```

Comment:
I realize now from this example `Atoms(symbols='Al12', pbc=[True, True, False], cell=[4.05, 4.05, 0.0])` that this is not correct. You should check that the cell is [0. 0. 0.]. The cell above should not set the `self.cell` to the `_DEFAULT_CELL`...

This raises the issue again of how to properly extend the concepts here to 1D and 2D